test/kubemark/start-kubemark.sh

```diff
@@ -314,8 +346,16 @@ current-context: kubemark-context")
   sed -i'' -e "s/{{EVENTER_MEM}}/${eventer_mem}/g" "${RESOURCE_DIRECTORY}/addons/heapster.json"
   "${KUBECTL}" create -f "${RESOURCE_DIRECTORY}/addons" --namespace="kubemark"
 
+  # When using Cluster Autoscaler, always start with one replica.
+  # NUM_NODES is the maximum number of nodes in the cluster and is used to
+  # calculate the resources for master and addons to be able to handle it.
+  if [[ "${ENABLE_KUBEMARK_CLUSTER_AUTOSCALER}" == "true" ]]; then
+    NUM_REPLICAS=1
+  else
+    NUM_REPLICAS=${NUM_NODES:-10}
```

Comment:
This doesn't seem right that we're setting a hard value of 1 inside the code when CA is enabled. Why can't you just set NUM_NODES=1 for the purpose of your tests instead of setting an unchangeable value like this?

Reply:
The problem is, Cluster Autoscaler will not work well if replicas in hollow node template is set to anything other than 1. We need to have 1 pod per controller, as we have to have an option to specify which specific pod to delete.
I have no problem with setting NUM_NODES=1 only for our test purposes, but this allows to have a partly broken setup.

Reply:
In that case, could you please move this out of the code? I'd suggest setting NUM_NODES=1 if CA is enabled inside cluster/kubemark/gce/config-default.sh.
Also note that we're using the NUM_NODES variable at other places too (for e.g in `create-master-environment-file` function).

Reply:
Good point. Also, I was not precise in my previous comment, I actually want NUM_NODES != NUM_REPLICAS. NUM_REPLICAS=1, NUM_NODES=max nodes that can be in the cluster (to make sure kubemark master is big enough). Changed this a bit, PTAL.

Reply:
Removed the defaults for NUM_NODES, rewarded the comment in cluster/kubemark/gce.config-default.sh